aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -22,12 +22,42 @@ using namespace at::sparse;
 
 namespace {
 
-  inline SparseTensor get_result_tensor_for_unary_op(const SparseTensor& input) {
-    if (c10::isIntegralType(input.scalar_type(), /*includeBool=*/true)) {
+  inline SparseTensor get_result_tensor_for_unary_op(const SparseTensor& input, bool promotes_integer_to_float = false) {
+    if (c10::isIntegralType(input.scalar_type(), /*includeBool=*/true) && promotes_integer_to_float) {
       return at::empty_like(input, input.options().dtype(c10::get_default_dtype()));
     }
     return at::empty_like(input);
   }
+
+  // --------------------------------------------------------------------
+  // helper function for sanity check of input and output tensors
+  // this functions also inits the output tensor
+  // this helper function is for unary ops
+  // --------------------------------------------------------------------
+  static inline void sparse_unary_op_check_and_init(SparseTensor& result, const SparseTensor& self,
+      bool promotes_integer_to_float = false, bool is_linear_map = false) {
```

Comment:
Defining
```
template <bool promotes_integer_to_float = false, bool is_linear_map = false>
void sparse_unary_op_check_and_init(SparseTensor& result, const SparseTensor& self)
```
might be more efficient in terms of a compilation result.

Reply:
@pearu Have done the changes as requested